src/lightning/data/streaming/functions.py

```diff
@@ -161,7 +175,8 @@ def map(
                 " HINT: You can either use `/teamspace/s3_connections/...` or `/teamspace/datasets/...`."
             )
 
-        _assert_dir_is_empty(output_dir)
+        if break_when_non_empty:
+            _assert_dir_is_empty(output_dir)
```

Comment:
Would you consider adding a test for this new argument?

Reply:
done !